juju/juju
# feat: storage registry dependency

This creates storage registry dependency for the domain services. The storage registry is required by various services, currently it is just passed as an argument to each service in the domain service at the API server level. This is incorrect and an anti-pattern. Instead, we want the storage registry to have its own lifetime independently of the domain services and the apiserver. The worker is a very light touch, it's just enough to get it up and running. The storage registry provider is correctly composed with the common storage providers (loop, rootfs, and tempfs). As this sits upon a provider tracker it can keep up to date automatically with any changes with the underlying provider.

The time with out-of-band direct access to the provider is over.

## QA steps

Ensure that we can bootstrap. Nothing uses this yet, so we just need to ensure that we haven't broken any dependencies.

```sh
$ juju bootstrap lxd test
$ juju add-model default
$ juju deploy ubuntu
```
